include/hwloc/rsmi.h

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright © 2012-2020 Inria.  All rights reserved.
+ * Copyright (c) 2020, Advanced Micro Devices, Inc. All rights reserved.
+ * Written by Advanced Micro Devices,
+ * See COPYING in top-level directory.
+ */
+
+/** \file
+ * \brief Macros to help interaction between hwloc and the AMD RSMI Management Library.
+ *
+ * Applications that use both hwloc and the RSMI Management Library may want to
+ * include this file so as to get topology information for AMD GPU devices.
+ */
+
+#ifndef HWLOC_RSMI_H
+#define HWLOC_RSMI_H
+
+#include "hwloc.h"
+#include "hwloc/autogen/config.h"
+#include "hwloc/helper.h"
+#ifdef HWLOC_LINUX_SYS
+#include "hwloc/linux.h"
+#endif
+
+#include <rocm_smi.h>
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+/** \defgroup hwlocality_rsmi Interoperability with the AMD RSMI Management Library
+ *
+ * This interface offers ways to retrieve topology information about
+ * devices managed by the RSMI Management Library.
+ *
+ * @{
+ */
+
+/**
+ * PCI information about a GPU device.
+ */
+
+typedef struct rsmiPciInfo_st {
+  union {
+    struct {
+#ifdef HWLOC_BIGENDIAN
+      uint64_t domain : 32;
+      uint64_t reserved : 16;
+      uint64_t bus : 8;
+      uint64_t device : 5;
+      uint64_t function : 3;
+#else
+      uint64_t function : 3;
+      uint64_t device : 5;
+      uint64_t bus : 8;
+      uint64_t reserved : 16;
+      uint64_t domain : 32;
+#endif
+    };
+    uint64_t bdfid;
+  };
+} rsmiPciInfo_t;
```

Comment:
If we keep this structure in hwloc as discussed above, remove it from topology-rsmi.c and #include hwloc/rsmi.h in topology-rsmi.c
Also we'll need HWLOC_BIGENDIAN to be exported to the *public* autogen/config.h since application will need the #define when they'll use hwloc/rsmi.h.
(I can take care of HWLOC_BIGENDIAN).

Reply:
Will use decode